sherpa/data.py

```diff
@@ -1580,10 +1717,30 @@ def get_xlabel(self):
         """
         return 'x'
 
-    def get_dims(self, filter=False):
+    # The superclass suggests returning both the independent and
+    # dependent axis sizes.
+    #
+    def get_dims(self, filter: bool = False) -> tuple[int]:
         return len(self.get_x(filter)),
 
-    def get_y(self, filter=False, yfunc=None, use_evaluation_space=False):
+    @overload
+    def get_y(self,
+              filter: bool,
+              yfunc: None = None,
+              use_evaluation_space: bool = False
+              ) -> np.ndarray:
+        ...
+
+    @overload
+    def get_y(self,
+              filter: bool,
+              yfunc: ModelFunc,
+              use_evaluation_space: bool = False
+              ) -> tuple[np.ndarray, ArrayType]:
```

Comment:
I suggest to also update the docstring to mention the second signature

Reply:
Done